src/library/scala/util/Random.scala

```diff
@@ -188,14 +188,20 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    *  @param  length    the desired length of the String
    *  @return           the String
    */
-  def nextString(length: Int) = {
-    def safeChar() = {
+  def nextString(length: Int): String = {
+    def safeChar(): Char = {
       val surrogateStart: Int = 0xD800
       val res = nextInt(surrogateStart - 1) + 1
       res.toChar
     }
-
-    List.fill(length)(safeChar()).mkString
+    val _length = java.lang.Math.max(length, 0)
+    val arr = new Array[Char](_length)
+    var i = 0
+    while (i < length) {
+      arr(i) = safeChar()
+      i += 1
+    }
```

Comment:
I'm happy to use the while loop here as it be slightly faster than `Array.fill(n)` which is generic and incurs some overhead.

Reply:
😄

Reply:
How about `Arrays.fill`?